core/src/main/scala/akka/kafka/internal/CommittingProducerSinkStage.scala

```diff
@@ -160,25 +161,25 @@ private final class CommittingProducerSinkStageLogic[K, V, IN <: Envelope[K, V,
     scheduleOnce(CommittingProducerSinkStage.CommitNow, stage.committerSettings.maxInterval)
 
   override protected def onTimer(timerKey: Any): Unit = timerKey match {
-    case CommittingProducerSinkStage.CommitNow => commit(Interval)
+    case CommittingProducerSinkStage.CommitNow => commit(Interval, flush = false)
   }
 
   private def collectOffset(count: Int, offset: Committable): Unit = {
     awaitingProduceResult -= count
     offsetBatch = offsetBatch.updated(offset)
-    if (offsetBatch.batchSize >= stage.committerSettings.maxBatch) commit(BatchSize)
-    else if (isClosed(stage.in) && awaitingProduceResult == 0L) commit(UpstreamClosed)
+    if (offsetBatch.batchSize >= stage.committerSettings.maxBatch) commit(BatchSize, flush = false)
+    else if (isClosed(stage.in) && awaitingProduceResult == 0L) commit(UpstreamClosed, flush = false)
   }
 
-  private def commit(triggeredBy: TriggerdBy): Unit = {
+  private def commit(triggeredBy: TriggerdBy, flush: Boolean): Unit = {
     if (offsetBatch.batchSize != 0) {
       log.debug("commit triggered by {} (awaitingProduceResult={} awaitingCommitResult={})",
                 triggeredBy,
                 awaitingProduceResult,
                 awaitingCommitResult)
       val batchSize = offsetBatch.batchSize
       offsetBatch
-        .commitInternal()
+        .commitInternal(flush)
```

Comment:
Maybe use the trigger instead to be more clear on when it is flushing:
```suggestion
        .commitInternal(flush = TriggeredBy == UpstreamFailure)
```

Reply:
sounds good